### ISSUE TITLE
Typed lambda parameter nullability not respected in field initializers for generic functional interfaces

### ISSUE THREAD

**Issue by HildingLinden** [uber/NullAway#1448]
Apologies if this is a duplicate of any of the existing lambda-related issues.

When implementing a generic functional interface whose method parameter is declared `@Nullable T`, NullAway reports a false positive if the lambda parameter is explicitly typed and annotated `@Nullable`, and the lambda is assigned to a field.

The same lambda passes when the parameter type is omitted and inferred, or when the lambda is assigned to a method-local variable.

This behavior occurs both with JSpecify mode enabled and disabled, and the false positive only occurs on Java 24+.

NullAway version: 0.13.0
Error Prone version: 2.43.0 
Java version: 24

Reproduction:
```java
package org.example;

import org.jspecify.annotations.NullMarked;
import org.jspecify.annotations.Nullable;

@NullMarked
interface Func<T> {
    void apply( @Nullable T t );
}

@NullMarked
class Bar {
    Func<Integer> baz = ( @Nullable Integer i ) -> {};
}
```

Warning:
```
Main.java:13: warning: [NullAway] parameter i is @NonNull, but parameter in functional interface method org.example.Func.apply(@org.jspecify.annotations.Nullable T) is @Nullable
    Func<Integer> baz = ( @Nullable Integer i ) -> {};
                                            ^
```
